Makefile

```diff
@@ -3,17 +3,39 @@
 
 ROOT_DIR=${CURDIR}
 LLVM_PROJ_DIR?=$(ROOT_DIR)/src/llvm-project
+
+# Windows needs munging
+ifeq ($(OS),Windows_NT)
+
+PREFIX?=c:/wasi-sdk
+# we need to explicitly call bash -c for makefile $(shell ...), otherwise we'll try under
+# who knows what
+BASH?=bash -c
+
+ifneq (x$(MSYSTEM),x)
```

Comment:
Do we support building on windows outside of MSYS?   How would that work?

Reply:
If not, can we just error here MSYSTEM is not set?

Reply:
We definitely don't right now.. I mean I guess we could as long as make is running in an environment where you can run bash, but yeah, that's a good idea.  Let's not make it easy for someone to do something that has a low chance of working.